fe/fe-core/src/main/java/com/starrocks/sql/analyzer/CTASAnalyzer.java

```diff
@@ -0,0 +1,143 @@
+// This file is licensed under the Elastic License 2.0. Copyright 2021 StarRocks Limited.
+package com.starrocks.sql.analyzer;
+
+import com.google.common.collect.Lists;
+import com.google.common.collect.Maps;
+import com.starrocks.analysis.ColumnDef;
+import com.starrocks.analysis.CreateTableAsSelectStmt;
+import com.starrocks.analysis.CreateTableStmt;
+import com.starrocks.analysis.DistributionDesc;
+import com.starrocks.analysis.Expr;
+import com.starrocks.analysis.HashDistributionDesc;
+import com.starrocks.analysis.InsertStmt;
+import com.starrocks.analysis.QueryStmt;
+import com.starrocks.analysis.SelectStmt;
+import com.starrocks.analysis.SlotRef;
+import com.starrocks.analysis.TableRef;
+import com.starrocks.analysis.TypeDef;
+import com.starrocks.catalog.Catalog;
+import com.starrocks.catalog.OlapTable;
+import com.starrocks.catalog.ScalarType;
+import com.starrocks.catalog.Table;
+import com.starrocks.catalog.Type;
+import com.starrocks.qe.ConnectContext;
+import com.starrocks.sql.analyzer.relation.QueryRelation;
+import com.starrocks.sql.analyzer.relation.Relation;
+import com.starrocks.sql.optimizer.statistics.ColumnStatistic;
+import com.starrocks.sql.optimizer.statistics.StatisticStorage;
+
+import java.util.List;
+import java.util.Map;
+
+public class CTASAnalyzer {
+    private final Catalog catalog;
+    private final ConnectContext session;
+
+    public CTASAnalyzer(Catalog catalog, ConnectContext session) {
+        this.catalog = catalog;
+        this.session = session;
+    }
+
+    public Relation transformCTASStmt(CreateTableAsSelectStmt createTableAsSelectStmt) {
+
+        List<String> columnNames = createTableAsSelectStmt.getColumnNames();
+        QueryStmt queryStmt = createTableAsSelectStmt.getQueryStmt();
+        CreateTableStmt createTableStmt = createTableAsSelectStmt.getCreateTableStmt();
+
+        QueryRelation queryRelation = new QueryAnalyzer(catalog, session)
+                .transformQueryStmt(queryStmt, new Scope(RelationId.anonymous(), new RelationFields()));
+
+        Map<String, Table> columnNameToTable = Maps.newHashMap();
+        Map<String, Table> tableRefToTableMap = Maps.newHashMap();
```

Comment:
I think it could update 'tableRefToTableMap' to 'tableRefToTable'.  Keep with 'columNameToTable'